server/controller/schedule.js

```diff
@@ -0,0 +1,78 @@
+import { check, validationResult } from "express-validator";
+// MODELS
+import Chore from "../models/Chore.js";
+import User from "../models/User.js";
+
+/*
+ ** This function creates a new schedule for the given date
+ ** @param {Object} req - The request object
+ ** @param {Object} res - The response object
+ */
+ export const createSchedule = async (req, res) => {
+
+    // Sanitize data
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+        // by default, time requested is set to the current time (which will return today's schedule)
+        let timeframeInMs = Date.now();
+
+        // if api req contains timeframe, then schedule for that date will be generated
+        if (req.query !== {}) {
+            //extracting timeframe from the request and converting it into milliseconds
+            timeframeInMs = Date.parse(req.query.timeframe);
+        }
+
+        var requestedSchedule = []; // will add chores for the day in here
+
+        // get the user that is requesting the schedule
+        const userID = req.user.id;
+        const user = await User.findOne({ _id: userID }); // search for the user
+        const choreList = user.chores;
+        
+        console.log("Printing chore ids for this user")
+
+        for (const choreId of choreList) {
+            // console.log(choreId);
+            const chore = await Chore.findOne({ _id: choreId });
```

Comment:
I don't think we need to be running a query against the DB for every chore in the list. `Chore.find( { _id: { $in: user.chores } });` automatically finds all chores with an ID in the user's list of chore IDs.

I think you should be able to put that line outside of the loop, populating some variable with the chore objects so we can run one query to get all the chores, then do whatever needs to be done.

Reply:
Done